mcp: add icon + website for testing SEP-973

jesselumarie opened the pull request:
## Motivation and Context
Adds a icon for the `simpleStreamableHttp.ts` server to test out SEP-973

## How Has This Been Tested?
Tested with the Figma MCP Server which has icons
<img width="566" height="166" alt="image" src="https://github.com/user-attachments/assets/34f8836c-339a-49b2-9331-477f21c2bacf" />


## Breaking Changes
n/a

## Types of changes
<!-- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update

## Checklist
<!-- Go over all the following points, and put an `x` in all the boxes that apply. -->
- [x] I have read the [MCP Documentation](https://modelcontextprotocol.io)
- [x] My code follows the repository's style guidelines
- [x] New and existing tests pass locally
- [x] I have added appropriate error handling
- [x] I have added or updated documentation as needed

cliffhall commented:
Hi @jesselumarie!

This looks cool. Just converted it to draft until the SEP is accepted and a Typescript SDK release contains the changes needed.

cliffhall commented:
The holdup on icons being an array instead of a space-separated list in a string [has been merged](https://github.com/modelcontextprotocol/typescript-sdk/pull/947/files#diff-c54113cf61ec99691748a3890bfbeb00e10efb3f0a76f03a0fd9ec49072e410aR242) in the SDK and is in the latest release.

@jesselumarie can you resolve the conflicts?

jesselumarie commented:
@cliffhall just pushed up a resolved version 👍🏻

cliffhall commented:
Hi @jesselumarie 

Your description section says 
> Adds a icon for the `simpleStreamableHttp.ts` server to test out SEP-973

But that would need to be done in the Typescript SDK, not the Inspector.

However I do see that the `simpleStreamableHttp.ts` server does in fact have an icon now. 
<img width="747" height="270" alt="Screenshot 2025-10-21 at 1 41 26 PM" src="https://github.com/user-attachments/assets/07f10c05-4da6-4407-b7e4-9bee8d5306da" />

But then you show a screenshot of a panel for the Figma server. I cannot tell where this panel appears within the UI or if it's a popup or what. 

When I run your branch right now, I have to update the Typescript SDK or it throws an error about the icons needing to be an array instead of a string. This has been fixed in the SDK and I have a [PR to bump the Inspector to the latest version of the SDK](https://github.com/modelcontextprotocol/inspector/pull/883) pending. Otherwise you see this when connecting to the `simpleStreamableHttp.ts` server:
<img width="739" height="578" alt="Screenshot 2025-10-21 at 1 45 09 PM" src="https://github.com/user-attachments/assets/562e9adb-fec7-4b0d-8106-ab5765f11e7b" />

With the latest version TS SDK, I see this when I connect to the `simpleStreamableHttp.ts` server.
<img width="1920" height="1115" alt="Screenshot 2025-10-21 at 1 43 06 PM" src="https://github.com/user-attachments/assets/32bbf9ad-78c6-464c-a923-3bf327041cfd" />

cliffhall commented on client/src/components/ToolsTab.tsx:
```diff
@@ -175,6 +177,17 @@ const ToolsTab = ({
+              <h3 className="font-semibold">
+                {selectedTool ? selectedTool.name : "Select a tool"}
+              </h3>
+            </div>
```
This section is leading to a doubling up of the tool name or "select a tool" message.

<img width="302" height="208" alt="Image" src="https://github.com/user-attachments/assets/6c4301db-725a-45ac-ae3c-462ed88dcde9" />

<img width="451" height="228" alt="Image" src="https://github.com/user-attachments/assets/dc8f2458-6533-4e5a-a173-4c49c5e8915e" />

jesselumarie commented:
@cliffhall i updated resources so they will show icons too, as well as removed the duplicate text issue. you can test it with this server claude spun up for me https://github.com/jesselumarie/mcp-icon-test-server. also updated the package.json to have the updated sdk.

cliffhall reviewed (changes requested):
@jesselumarie Tested with your example server that has icons for everything and it's looking good and ready to merge. 

However, I'm not certain we need these changes to `package-lock.json`; there was no `package.json` change.

jesselumarie commented:
@cliffhall i think that's better — holler if you see any other issues

cliffhall reviewed (approved):
LGTM! 👍
